### ISSUE TITLE
CVE-2022-0235 - node-fetch (peer dependency via isomorphic-fetch)

### ISSUE THREAD

**Issue by nathanlodge** [avadev/AvaTax-REST-V2-JS-SDK#213]
### CVE-2022-0235

Versions of the `node-fetch` package lower than 2.6.7 are potentially vulnerable to exposure of sensitive information to an unauthorized actor (see https://github.com/advisories/GHSA-r683-j2x4-v87g)

The `node-fetch` package is a peer dependency of the `Avatax-REST-V2-JS-SDK` package via the `isomorphic-fetch` package:

https://github.com/avadev/AvaTax-REST-V2-JS-SDK/blob/475c6a69243194edf123338799853a246b0a07ba/package.json#L26

https://github.com/avadev/AvaTax-REST-V2-JS-SDK/blob/475c6a69243194edf123338799853a246b0a07ba/yarn.lock#L1656-L1661

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ node-fetch is vulnerable to Exposure of Sensitive            │
│               │ Information to an Unauthorized Actor                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ node-fetch                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.6.7                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ isomorphic-fetch                                             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ isomorphic-fetch > node-fetch                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1006899                     │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

### Overriding `node-fetch` to a safe version is not currently an option

Testing with `node-fetch` version `2.6.7`, the following TypeError is encountered `TypeError: Cannot read properties of undefined (reading 'content-type')` handling responses here:

https://github.com/avadev/AvaTax-REST-V2-JS-SDK/blob/475c6a69243194edf123338799853a246b0a07ba/lib/AvaTaxClient.js#L88-L89

Would it be possible to prioritize a fix for this vulnerability?

**Comment by RodneyMorgan97** [comment]
I'm attempting to leverage the `client.resolveAddressPost(address);` function in NextJS (Typescript) and am also receiving `TypeError: Cannot read properties of undefined (reading 'content-type')` on response. 

Prioritizing fix for this would be much appreciated!

**Comment by eboureau** [comment]
My PR #214 should address this issue

**Comment by ui-artur-zapater** [comment]
Is there any estimate as to when that PR will be merged?

**Comment by nathanlodge** [comment]
Resolved by https://github.com/avadev/AvaTax-REST-V2-JS-SDK/commit/7d87bb477fc2998655affecf53541aa1902bd044

**Comment by chelevich** [comment]
Should it make the package version bumped to 22.3.1?

**Comment by chelevich** [comment]
Resolved in 22.5.0
